src/components/Header.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { TodoContext } from '../context/TodoContext';
+import { Error } from '../types/ErrorMessage';
+import * as TodoService from '../api/todos';
+import { USER_ID } from '../types/userId';
+
+export const Header: React.FC = () => {
+  const [newTitle, setNewTitle] = useState('');
+  const [disableInput, setDisableInput] = useState(false);
+
+  const {
+    todos,
+    tempTodo,
+    addTodo,
+    updateTodo,
+    handleSetTempTodo,
+    handleSetErrorMessage,
+    handleSetUpdatingIds,
```

Comment:
```suggestion
    setTempTodo,
    setErrorMessage,
    setUpdatingIds,
```

Reply:
Ok, now I use just setTempTodo and setErrorMessage (setWHATEVER: React.Dispatch<React.SetStateAction<type>>). but I have to keep SetUpdatingTodos via 

const SetUpdatingIds = (id: number | null) => {
    if (id) {
      setUpdatingIds(currentIds => [...currentIds, id]);
    } else {
      setUpdatingIds([]);
    }
  };    
  
  
  for implementation two paths:

1)passing a number, which gets added to the array.
2)setting the array to empty.

please let me know if there is a better way - I'll do it.